dotnet-desktop-guide/winforms/migration/snippets/clipboard-dataobject-net10/net/csharp/ModernApproach.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.Windows.Forms;
+
+namespace ClipboardExamples
+{
+    public class ModernApproach
+    {
+        public class Person
+        {
+            public string Name { get; set; }
+            public int Age { get; set; }
+        }
+
+        // <ModernTryGetData>
+        public static void ModernTryGetDataExample()
+        {
+            // Use this - type-safe approach with TryGetData<T>()
+            if (Clipboard.TryGetData("MyApp.Person", out Person person))
```

Comment:
This code isn't particularly helpful without the set/get loop. The true modern way would be to `SetDataAsJson` then `TryGetData`.